javascript/canvas.js

```diff
@@ -2,33 +2,65 @@ class HangmanCanvas {
   constructor(secretWord) {
     this.context = document.getElementById('hangman').getContext('2d');
     // ... your code goes here
+    this.secretWord = secretWord
   }
 
   createBoard() {
     // ... your code goes here
+    this.context.clearRect(0,0, 1200, 800)
+    
   }
 
   drawLines() {
     // ... your code goes here
+    //this.secretWord.split("").forEach((letter,i) => this.drawHangman()
+    this.context.linewidth = 10;
+    for (let i = 0; i < this.secretWord.length; i++) {
+      this.context.beginPath();
+      this.context.moveTo(10 + (i * 50), 750);
+      this.context.lineTo(50 + (i * 50), 750);
+      this.context.stroke();
+    }
+    //link https://dirask.com/posts/JavaScript-draw-line-on-canvas-element-Z1A7ap
   }
 
   writeCorrectLetter(index) {
     // ... your code goes here
+    this.context.fillStyle = "black";
+    this.context.font = '35px Arial'
+
+   let letter =  this.secretWord[index].toUpperCase()
+   //console.log(letter, "palabra screta:", this.secretWord)
+   for(let i = 0; i < this.secretWord.length; i++){
+     if(this.secretWord[i].toUpperCase() === letter){
+       this.context.fillText(letter,260 + 50 * i + 10 * i , 680, 50)
+     }
+   }
   }
 
   writeWrongLetter(letter, errorsLeft) {
     // ... your code goes here
+    const wrongLetter = 10 - errorsLeft;
+    this.context.fillStyle = "Red"
+    this.context.font = "35px Arial"
+    if (!this.secretWord.includes(letter)) {
+      this.context.fillText(letter.toUpperCase(),500 + (60 * errorsLeft), 100)
+    }
   }
 
   drawHangman(errorsLeft) {
     // ... your code goes here
+    //swich y 9 casos
```

Comment:
`  switch (errorsLeft) {
      case 9:
        this.context.beginPath();
        this.context.arc(500, 225, 50, 0, Math.PI * 2);
        this.context.closePath();
        this.context.stroke();
        break;
      case 8:
        this.context.moveTo(500, 275);
        this.context.lineTo(500, 425);
        this.context.stroke();
        break;
      case 7:
        this.context.lineTo(450, 500);
        this.context.stroke();
        break;
      case 6:
        this.context.lineTo(420, 500);
        this.context.stroke();
        break;
      case 5:
        this.context.moveTo(500, 425);
        this.context.lineTo(550, 500);
        this.context.stroke();
        break;
      case 4:
        this.context.lineTo(580, 500);
        this.context.stroke();
        break;
      case 3:
        this.context.moveTo(500, 330);
        this.context.lineTo(430, 330);
        this.context.stroke();
        break;
      case 2:
        this.context.lineTo(570, 330);
        this.context.stroke();
        break;
      case 1:
        this.context.beginPath();
        this.context.arc(485, 210, 5, 0, Math.PI * 2);
        this.context.closePath();
        this.context.stroke();
        this.context.beginPath();
        this.context.arc(515, 210, 5, 0, Math.PI * 2);
        this.context.closePath();
        this.context.stroke();
        this.context.moveTo(485, 245);
        this.context.lineTo(515, 245);
        this.context.stroke();
        break;
      case 0: // uh oh
        this.context.font = '30px Arial';
        this.context.clearRect(470, 190, 60, 40);
        this.context.fillText('x', 479, 217);
        this.context.fillText('x', 509, 217);
        break;
    }`  este podria ser un caso cerca pero solo cambian los valores de las x y Y por cuestion de tu dibujo y donde pones las letras cuando las agregas